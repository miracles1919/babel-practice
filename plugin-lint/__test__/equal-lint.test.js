const fs = require('fs');
const path = require('path');
const pluginTester = require('babel-plugin-tester').default;

const EqualLint = require('../plugin/equal-lint');

const source = fs
  .readFileSync(path.join(__dirname, '../source/equal.js'))
  .toString();

pluginTester({
  plugin: EqualLint,
  pluginName: 'EqualLint',
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous',
    },
  },
  tests: {
    default: {
      code: source,
      snapshot: true,
    },

    'auto fix': {
      code: source,
      snapshot: true,
      pluginOptions: {
        fix: true,
      },
    },
  },
});
