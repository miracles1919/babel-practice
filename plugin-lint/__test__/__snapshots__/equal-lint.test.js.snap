// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EqualLint auto fix: auto fix 1`] = `

'' == false;
0 == false;
false == false;
'' != false;

'' === false;
0 === false;

      ↓ ↓ ↓ ↓ ↓ ↓

"" === false;
0 === false;
false == false;
"" !== false;
"" === false;
0 === false;


`;

exports[`EqualLint default: default 1`] = `

'' == false;
0 == false;
false == false;
'' != false;

'' === false;
0 === false;

      ↓ ↓ ↓ ↓ ↓ ↓

"" == false;
0 == false;
false == false;
"" != false;
"" === false;
0 === false;


`;
